--- /Users/adamknowles/.platformio/packages/framework-arduinoespressif32/libraries/WebServer/src/WebServer.cpp	2024-06-06 13:16:55
+++ lib/WebServer/WebServer.cpp	2025-08-12 23:33:00
@@ -421,7 +421,7 @@
     // Can we asume the following?
     //if(code == 200 && content.length() == 0 && _contentLength == CONTENT_LENGTH_NOT_SET)
     //  _contentLength = CONTENT_LENGTH_UNKNOWN;
-    if (content.length() == 0) {
+    if (content.length() == 0 && code != 304 && code != 204) {
         log_w("content length is zero");
     }
     _prepareHeader(header, code, content_type, content.length());
